File: snuggle_server/src/main/java/com/ssafy/snuggle/model/service/NotificationService.java
package com.ssafy.snuggle.model.service;

import java.util.List;

import com.ssafy.snuggle.model.dto.Notification;

public interface NotificationService {

	/**
     * 사용자 id에 따른 알림 리스트 조회
     * @param userId
     */
	List<Notification> getNotificationByUserId(String userId);
	
	/**
     * 알림 발생 시, 알림 리스트에 추가
     * @param Notification
     */
	int addNotification(Notification notice);
	
}

File: snuggle_final_app/app/src/main/java/com/ssafy/snuggle_final_app/data/service/NotificationService.kt
package com.ssafy.snuggle_final_app.data.service

import com.ssafy.snuggle_final_app.data.model.dto.Notification
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface NotificationService {

    @POST("/snuggle/notification")
    suspend fun addNotification(@Body notification: Notification): Response<Int>

    @GET("/snuggle/notification")
    suspend fun getNotificationByUserId(@Query("userId") userId: String) : Response<List<Notification>>
}
